game/game/playercontrol.cpp

```diff
@@ -196,12 +198,13 @@ void PlayerControl::tickFocus() {
   currentFocus = findFocus(&currentFocus);
   setTarget(currentFocus.npc);
 
-  if(!ctrl[Action::ActionGeneric])
+  if(!ctrl[Action::ActionGeneric] || Gothic::inst().world()->player()->interactive()!=nullptr)
     return;
 
   auto focus = currentFocus;
   if(focus.interactive!=nullptr && interact(*focus.interactive)) {
-    clearInput();
+    if (!focus.interactive->isLadder())
+      clearInput();
```

Comment:
What is the intent of this code? sorry, This related to key-repeat stuff, right?

Reply:
It's for G1. In vanilla, if you interact with ladder and keep action key pressed you stay attached to ladder. Without this change you are released.